src/main/scala/io/iohk/ethereum/network/p2p/validators/BlockHeaderValidator.scala

```diff
@@ -0,0 +1,163 @@
+    *
+    * @param blockHeader BlockHeader to validate.
+    * @param parentHeader BlockHeader of the parent of the block to validate.
+    * @return BlockHeader if valid, an [[HeaderDifficultyError]] otherwise
+    */
+  private def validateDifficulty(blockHeader: BlockHeader, parentHeader: BlockHeader): Either[BlockHeaderError, BlockHeader] =
+    if(calculateDifficulty(blockHeader, parentHeader) == blockHeader.difficulty) Right(blockHeader)
+    else Left(HeaderDifficultyError)
+
+  /**
+    * Validates [[io.iohk.ethereum.domain.BlockHeader.gasUsed]] is not greater than [[io.iohk.ethereum.domain.BlockHeader.gasLimit]]
+    * based on validations stated in section 4.4.4 of http://paper.gavwood.com/
+    *
+    * @param blockHeader BlockHeader to validate.
+    * @return BlockHeader if valid, an [[HeaderGasUsedError]] otherwise
+    */
+  private def validateGasUsed(blockHeader: BlockHeader): Either[BlockHeaderError, BlockHeader] =
+    if(blockHeader.gasUsed<=blockHeader.gasLimit) Right(blockHeader)
+    else Left(HeaderGasUsedError)
+
+  /**
+    * Validates [[io.iohk.ethereum.domain.BlockHeader.gasLimit]] follows the restrictions based on its parent gasLimit
+    * based on validations stated in section 4.4.4 of http://paper.gavwood.com/
+    *
+    * @param blockHeader BlockHeader to validate.
+    * @param parentHeader BlockHeader of the parent of the block to validate.
+    * @return BlockHeader if valid, an [[HeaderGasLimitError]] otherwise
+    */
+  private def validateGasLimit(blockHeader: BlockHeader, parentHeader: BlockHeader): Either[BlockHeaderError, BlockHeader] = {
+    val gasLimitDiff = (blockHeader.gasLimit - parentHeader.gasLimit).abs
+    val gasLimitDiffLimit = parentHeader.gasLimit / GasLimitBoundDivisor
+    if(gasLimitDiff < gasLimitDiffLimit && blockHeader.gasLimit >= MinGasLimit) Right(blockHeader)
+    else Left(HeaderGasLimitError)
+  }
+
+  /**
+    * Validates [[io.iohk.ethereum.domain.BlockHeader.number]] is the next one after its parents number
+    * based on validations stated in section 4.4.4 of http://paper.gavwood.com/
+    *
+    * @param blockHeader BlockHeader to validate.
+    * @param parentHeader BlockHeader of the parent of the block to validate.
+    * @return BlockHeader if valid, an [[HeaderNumberError]] otherwise
+    */
+  private def validateNumber(blockHeader: BlockHeader, parentHeader: BlockHeader): Either[BlockHeaderError, BlockHeader] =
+    if(blockHeader.number == parentHeader.number + 1) Right(blockHeader)
+    else Left(HeaderNumberError)
+
+  /**
+    * Validates [[io.iohk.ethereum.domain.BlockHeader.nonce]] and [[io.iohk.ethereum.domain.BlockHeader.mixHash]] are correct
+    * based on validations stated in section 4.4.4 of http://paper.gavwood.com/
+    *
+    * @param blockHeader BlockHeader to validate.
+    * @return BlockHeader if valid, an [[HeaderPoWError]] otherwise
+    */
+  //FIXME: Simple PoW validation without using DAG
```

Comment:
[[EC-88]](https://iohk.myjetbrains.com/youtrack/issue/EC-88) added.... feel free to comment on it..